### ISSUE TITLE
matmul gemm and ip tests failing with  attribute --attr-scales

### ISSUE THREAD

**Issue by cfRod** [oneapi-src/oneDNN#1495]
# Summary
Matmul reference gemm implementation (https://github.com/oneapi-src/oneDNN/blob/master/src/cpu/matmul/gemm_f32_matmul.hpp#L40) is failing with the introduction of --attr-scales in benchdnn on aarch64.

# Version
oneDNN git hash commit 9b5e3dd98dbdffea6904fa08679648e3b04ab695 

# Environment
* CPU aarch64
* OS version Ubuntu 20.04.4
* Compiler version 10.3.0
* CMake version 3.24.1

# Steps to reproduce
with --attr-scale it fails
```
 DNNL_VERBOSE=1 ./tests/benchdnn/benchdnn  --matmul --stag=ab --wtag=ba --dtag=ab --bia_dt=f32 --attr-scales=src:common:0.25*+dst:common:2.25*+wei:per_oc:0.5* --attr-post-ops=relu 10x1:1x20 
onednn_verbose,info,oneDNN v3.0.0 (commit 9b5e3dd98dbdffea6904fa08679648e3b04ab695)
onednn_verbose,info,cpu,runtime:OpenMP,nthr:48
onednn_verbose,info,cpu,isa:AArch64 SVE (256 bits)
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:ab:f0 dst_f32::blocked:ab:f0,,,10x1,0.130859
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:ab:f0 dst_f32::blocked:ba:f0,,,1x20,0.0410156
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:ab:f0 dst_f32::blocked:ba:f0,,,1x20,0.00585938
onednn_verbose,exec,cpu,matmul,gemm:jit,undef,src_f32::blocked:ab:f0 wei_f32::blocked:ba:f0 bia_f32::blocked:ab:f0_mask2 dst_f32::blocked:ab:f0,attr-scales:src0:0+dst:0+wei:2 attr-post-ops:eltwise_relu ,,10x1:1x20:10x20,1.82397
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:ab:f0 dst_f32::blocked:ab:f0,,,10x20,0.0380859
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:ab:f0 dst_f32::blocked:ab:f0,,,10x20,0.00488281
[   0][DST][0:0] exp_f32: 54.0556 exp: 54.0556 got: 121.625 diff: 67.5694 rdiff:    1.25
[   1][DST][0:1] exp_f32: 12.3611 exp: 12.3611 got: 27.8125 diff: 15.4514 rdiff:    1.25
[   2][DST][0:2] exp_f32: 43.5833 exp: 43.5833 got: 98.0625 diff: 54.4792 rdiff:    1.25
[   3][DST][0:3] exp_f32: 64.0556 exp: 64.0556 got: 144.125 diff: 80.0694 rdiff:    1.25
[  10][DST][0:10] exp_f32: 807.556 exp: 807.556 got:    1817 diff: 1009.44 rdiff:    1.25
[  11][DST][0:11] exp_f32: 417.333 exp: 417.333 got:     939 diff: 521.667 rdiff:    1.25
[  12][DST][0:12] exp_f32: 912.889 exp: 912.889 got:    2054 diff: 1141.11 rdiff:    1.25
[  13][DST][0:13] exp_f32: 2059.56 exp: 2059.56 got:    4634 diff: 2574.44 rdiff:    1.25
[  14][DST][0:14] exp_f32: 706.222 exp: 706.222 got:    1589 diff: 882.778 rdiff:    1.25
0:FAILED (errors:96 total:200) __REPRO: --matmul --stag=ab --wtag=ba --dtag=ab --bia_dt=f32 --attr-scales=src:common:0.25*+dst:common:2.25*+wei:per_oc:0.5* --attr-post-ops=relu 10x1:1x20
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 invalid_arguments:0 failed:1 listed:0
total compute_ref: sum(s):0.00
```
Without --attr-scale, it passes
```
 DNNL_VERBOSE=1 ./tests/benchdnn/benchdnn  --matmul --stag=ab --wtag=ba --dtag=ab --bia_dt=f32 --attr-post-ops=relu 10x1:1x20 
onednn_verbose,info,oneDNN v3.0.0 (commit 9b5e3dd98dbdffea6904fa08679648e3b04ab695)
onednn_verbose,info,cpu,runtime:OpenMP,nthr:48
onednn_verbose,info,cpu,isa:AArch64 SVE (256 bits)
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:ab:f0 dst_f32::blocked:ab:f0,,,10x1,0.144043
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:ab:f0 dst_f32::blocked:ba:f0,,,1x20,0.187012
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:ab:f0 dst_f32::blocked:ba:f0,,,1x20,0.00683594
onednn_verbose,exec,cpu,matmul,gemm:jit,undef,src_f32::blocked:ab:f0 wei_f32::blocked:ba:f0 bia_f32::blocked:ab:f0_mask2 dst_f32::blocked:ab:f0,attr-post-ops:eltwise_relu ,,10x1:1x20:10x20,1.89795
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:ab:f0 dst_f32::blocked:ab:f0,,,10x20,0.0410156
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:ab:f0 dst_f32::blocked:ab:f0,,,10x20,0.00610352
0:PASSED __REPRO: --matmul --stag=ab --wtag=ba --dtag=ab --bia_dt=f32 --attr-post-ops=relu 10x1:1x20
tests:1 passed:1 skipped:0 mistrusted:0 unimplemented:0 invalid_arguments:0 failed:0 listed:0
total compute_ref: sum(s):0.00                                                    
```

**Comment by cfRod** [comment]
ip fail test example:
```
DNNL_VERBOSE=1 ./tests/benchdnn/benchdnn  --ip --dir=FWD_D --cfg=u8s8u8 --attr-scales=src:common:0.25*+dst:common:2.25*+wei:per_oc:0.5* --attr-post-ops=sum mb1000ic1oc16
onednn_verbose,info,oneDNN v3.0.0 (commit 9b5e3dd98dbdffea6904fa08679648e3b04ab695)
onednn_verbose,info,cpu,runtime:OpenMP,nthr:48
onednn_verbose,info,cpu,isa:AArch64 SVE (256 bits)
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:ab:f0 dst_u8::blocked:ab:f0,,,1000x1,0.00292969
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:ab:f0 dst_s8::blocked:ab:f0,,,16x1,0
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:ab:f0 dst_u8::blocked:ab:f0,,,1000x16,0.797119
onednn_verbose,exec,cpu,inner_product,gemm_s8u8s32:ref,forward_training,src_u8::blocked:ab:f0 wei_s8::blocked:ab:f0 bia_undef::undef::f0 dst_u8::blocked:ab:f0,attr-scales:src0:0+dst:0+wei:1 attr-post-ops:sum ,,mb1000ic1oc16,0.474854
onednn_verbose,exec,cpu,gemm_api,,undef,src_f32::blocked:ab:f0 wei_f32::blocked:ba:f0 dst_f32::blocked:ab:f0,,,1000x1:1x16,0.0341797
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_u8::blocked:ab:f0 dst_f32::blocked:ab:f0,,,1000x16,0.00683594
[   8][DST][0:8] exp_f32:     67.5556 exp:          68 got:         152 diff:      84 rdiff: 1.23529
[   9][DST][0:9] exp_f32:     33.3333 exp:          33 got:          75 diff:      42 rdiff: 1.27273
[  10][DST][0:10] exp_f32:     112.889 exp:         113 got:         254 diff:     141 rdiff: 1.24779
[  11][DST][0:11] exp_f32:     78.6667 exp:          79 got:         177 diff:      98 rdiff: 1.24051
[  12][DST][0:12] exp_f32:     44.4444 exp:          44 got:         100 diff:      56 rdiff: 1.27273
[  17][DST][1:1] exp_f32:     42.6111 exp:          43 got:          96 diff:      53 rdiff: 1.23256
[  18][DST][1:2] exp_f32:     8.58333 exp:           9 got:          19 diff:      10 rdiff: 1.11111
[  19][DST][1:3] exp_f32:     88.0556 exp:          88 got:         198 diff:     110 rdiff:    1.25
[  20][DST][1:4] exp_f32:     53.7361 exp:          54 got:         121 diff:      67 rdiff: 1.24074
0:FAILED (errors:8705 total:16000) __REPRO: --ip --dir=FWD_D --cfg=u8s8u8 --attr-scales=src:common:0.25*+dst:common:2.25*+wei:per_oc:0.5* --attr-post-ops=sum mb1000ic1oc16
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 invalid_arguments:0 failed:1 listed:0
```
without --attr-scale:
```
DNNL_VERBOSE=1 ./tests/benchdnn/benchdnn  --ip --dir=FWD_D --cfg=u8s8u8  --attr-post-ops=sum mb1000ic1oc16
onednn_verbose,info,oneDNN v3.0.0 (commit 9b5e3dd98dbdffea6904fa08679648e3b04ab695)
onednn_verbose,info,cpu,runtime:OpenMP,nthr:48
onednn_verbose,info,cpu,isa:AArch64 SVE (256 bits)
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:ab:f0 dst_u8::blocked:ab:f0,,,1000x1,0.00317383
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:ab:f0 dst_s8::blocked:ab:f0,,,16x1,0.0012207
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:ab:f0 dst_u8::blocked:ab:f0,,,1000x16,0.846924
onednn_verbose,exec,cpu,inner_product,gemm_s8u8s32:ref,forward_training,src_u8::blocked:ab:f0 wei_s8::blocked:ab:f0 bia_undef::undef::f0 dst_u8::blocked:ab:f0,attr-post-ops:sum ,,mb1000ic1oc16,0.641113
onednn_verbose,exec,cpu,gemm_api,,undef,src_f32::blocked:ab:f0 wei_f32::blocked:ba:f0 dst_f32::blocked:ab:f0,,,1000x1:1x16,0.165771
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_u8::blocked:ab:f0 dst_f32::blocked:ab:f0,,,1000x16,0.00708008
0:PASSED __REPRO: --ip --dir=FWD_D --cfg=u8s8u8 --attr-post-ops=sum mb1000ic1oc16
tests:1 passed:1 skipped:0 mistrusted:0 unimplemented:0 invalid_arguments:0 failed:0 listed:0
total compute_ref: sum(s):0.00
```

**Comment by vpirogov** [comment]
Fixed by cb7eeb3d597183a8d66114429811843612d0e82a in master and 0f3c5ca8e33a4368eeb86326af3dfe6aab6eb401 in rls-v3.0.

**Comment by cfRod** [comment]
Thanks!
